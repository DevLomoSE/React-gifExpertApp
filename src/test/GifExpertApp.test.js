const { shallow } = require("enzyme");
const { default: GifExpertApp } = require("../GifExpertApp");

describe('Testing for component <GifExpertApp />', () => {

    test('should render', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should list of categories', () => {
        const categories = ['lord of the rings', 'the hobbit'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});