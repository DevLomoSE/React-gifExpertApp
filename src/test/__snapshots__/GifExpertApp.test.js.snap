// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for component <GifExpertApp /> should list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="lord of the rings"
      key="0"
    />
    <GifGrid
      category="the hobbit"
      key="1"
    />
  </ol>
</Fragment>
`;

exports[`Testing for component <GifExpertApp /> should render 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
